### PR TITLE
Fix MultipleLocator proposed by @anntzer

### PR DESCRIPTION
**Pull request by rougier**
This is the fix proposed by @anntzer in #7122 related to a bug in MultipleLocator ticker.

### CODE REVIEW

File: lib/matplotlib/ticker.py [hunk 1]
```diff
@@ -1614,10 +1614,10 @@ def __call__(self):
     def tick_values(self, vmin, vmax):
         if vmax < vmin:
             vmin, vmax = vmax, vmin
-        vmin = self._base.ge(vmin)
         base = self._base.get_base()
-        n = (vmax - vmin + 0.001 * base) // base
-        locs = vmin - base + np.arange(n + 3) * base
+        nmin = round(self._base.ge(vmin) / base)
+        nmax = round(self._base.le(vmax) / base)
+        locs = np.arange(nmin, nmax + 1) * base
```
Comment (NelleV) [inline lib/matplotlib/ticker.py]: Why can't this code be simplified to: `np.arange(vmin, vmax + base, base)? Floating point errors?

Comment (NelleV) [inline lib/matplotlib/ticker.py]: Sorry… np.arange(self._base.ge(vmin), vmax + base, base) would be the correct solution.

**Comment by NelleV** [comment]
I have done a couple of tests locally, and I have to admit that the current tests seems obscure…
@pelson you wrote this test. Can you comment on this? Some ticks are outside of the bounds provided by the function: it doesn't seem the correct behavior to me.

File: lib/matplotlib/ticker.py (hunk 1 above)
Comment (anntzer) [inline lib/matplotlib/ticker.py]: Your proposal can include a value that's above vmax (eg. vmin=.5, vmax=2.5, base=1 will return [1, 2, 3]).  I guess a solution that's safe against floating point errors is
```np.arange(self._base.ge(vmin), self._base.le(vmax) + base / 2, base)```
(the `+ base / 2` should take care of any fpe).

Comment (NelleV) [inline lib/matplotlib/ticker.py]: Yep: I am just concern about the tests that don't make sense to me.

Comment (anntzer) [inline lib/matplotlib/ticker.py]: I had a quick look and am pretty sure that the test values are wrong (i.e. they were picked to match the previous algorithm, which returns out of bounds values; they should be changed to match the new algorithm).

**Comment by tacaswell** [comment]
I think that this ties back to the other issue with assuming all of the ticks are in view limits (https://github.com/matplotlib/matplotlib/pull/6647) and what layer of the stack is responsible for the final filtering.

**Comment by tacaswell** [comment]
When this is cleared up please clean up the figure added in #7084

**Comment by jklymak** [comment]
This is pretty stale, so I'm assuming this change isn't urgently required by anyone, but feel free to reopen if wanted...

**Pull request closed without merging**